processor/normalizesumsprocessor/processor.go

```diff
@@ -0,0 +1,336 @@
+		)
+		return false
+	}
+
+	// If data has rolled over or the counter has been restarted for
+	// any other reason, grab a new start point and do not report this data
+	if dp.Value() < start.lastDoubleValue {
+		dp.CopyTo(*start.doubleDataPoint)
+		start.lastDoubleValue = dp.Value()
+
+		return false
+	}
+
+	start.lastDoubleValue = dp.Value()
+	dp.SetValue(dp.Value() - start.doubleDataPoint.Value())
+	dp.SetStartTimestamp(start.doubleDataPoint.Timestamp())
+
+	return true
+}
+
+func (nsp *NormalizeSumsProcessor) processIntSumDataPoint(dp pdata.IntDataPoint, resource pdata.Resource, metric pdata.Metric) bool {
+	metricIdentifier := dataPointIdentifier(resource, metric, dp.LabelsMap())
+
+	start := nsp.history[metricIdentifier]
+	// If this is the first time we've observed this unique metric,
+	// record it as the start point and do not report this data point
+	if start == nil {
+		dps := metric.IntSum().DataPoints()
+		newDP := pdata.NewIntDataPoint()
+		dps.At(0).CopyTo(newDP)
+
+		newStart := startPoint{
+			dataType:     pdata.MetricDataTypeIntSum,
+			intDataPoint: &newDP,
+			lastIntValue: newDP.Value(),
+		}
+		nsp.history[metricIdentifier] = &newStart
+
+		return false
+	}
+
+	// If this data is older than the start point, we can't meaningfully report this point
+	// TODO - consider resetting on two subsequent data points older than current start timestamp.
+	// This could signify a permanent clock change.
+	if dp.Timestamp() <= start.intDataPoint.Timestamp() {
+		nsp.logger.Info(
+			"data point being processed older than last recorded reset, will not be emitted",
+			zap.String("lastRecordedReset", start.intDataPoint.Timestamp().String()),
+			zap.String("dataPoint", dp.Timestamp().String()),
+		)
+		return false
+	}
+
+	// If data has rolled over or the counter has been restarted for
+	// any other reason, grab a new start point and do not report this data
+	if dp.Value() < start.lastIntValue {
+		dp.CopyTo(*start.intDataPoint)
+		start.lastIntValue = dp.Value()
+
+		return false
+	}
+
+	start.lastIntValue = dp.Value()
+	dp.SetValue(dp.Value() - start.intDataPoint.Value())
+	dp.SetStartTimestamp(start.intDataPoint.Timestamp())
+
+	return true
+}
+
+func dataPointIdentifier(resource pdata.Resource, metric pdata.Metric, labels pdata.StringMap) string {
+	var b strings.Builder
+
+	// Resource identifiers
+	resource.Attributes().Sort().Range(func(k string, v pdata.AttributeValue) bool {
+		fmt.Fprintf(&b, "%s=", k)
+		addAttributeToIdentityBuilder(&b, v)
+		b.WriteString("|")
+		return true
+	})
+
+	// Metric identifiers
+	fmt.Fprintf(&b, " - %s", metric.Name())
+	labels.Sort().Range(func(k, v string) bool {
+		fmt.Fprintf(&b, " %s=%s", k, v)
+		return true
+	})
+	return b.String()
+}
+
+func addAttributeToIdentityBuilder(b *strings.Builder, v pdata.AttributeValue) {
+	switch v.Type() {
+	case pdata.AttributeValueTypeArray:
+		b.WriteString("[")
+		arr := v.ArrayVal()
+		for i := 0; i < arr.Len(); i++ {
```

Comment:
FYI - We're likely to spec out this behavior so hopefully you get a helper method from the collector going forward.